python/neuroglancer/downsample.py

```diff
@@ -23,9 +23,21 @@ def method(layer_type):
     return downsample_with_averaging
   elif layer_type == 'segmentation':
     return downsample_segmentation
+  elif layer_type == 'activation':
+    return downsample_with_max_pooling
   else:
     return downsample_with_striding 
 
+def validate_factor(array, factor):
+  factor = np.array(factor, dtype=np.int32)
+  if np.any(factor <= 0):
+    raise ValueError("Factors less than one don't make sense. Factor: {}".format(factor))
+
+  if len(array.shape) == 4 and len(factor) == 3:
```

Comment:
Hmm, there's sort of a deeper reason for this. I'm not sure what the right answer is yet. If you can write code that's like (2,2,2) instead of (2,2,2,1), then you don't need to rewrite the code for (2,2,2,3) or (2,2,2,N) since they are all the same.